apps/charterafrica/src/components/EcosystemList/EcosystemFilter.js

```diff
@@ -0,0 +1,109 @@
+import { RichTypography } from "@commons-ui/core";
+import { Box, Grid, SvgIcon, TextField } from "@mui/material";
+import React from "react";
+
+import ChevronDown from "@/charterafrica/assets/icons/Type=chevron-down, Size=16, Color=CurrentColor.svg";
+import { neutral } from "@/charterafrica/colors";
+import ComboBox from "@/charterafrica/components/ComboBox/ComboBox";
+import SearchInput from "@/charterafrica/components/SearchInput";
+
+const EcosystemFilter = React.forwardRef(function EcosystemFilter(props, ref) {
+  const {
+    onChange,
+    searchPlaceholder,
+    values,
+    onQuerySearch,
+    filterOptions = [],
+  } = props;
+  const handleKeyPress = (e) => {
+    if (e.key === "Enter") {
+      if (onQuerySearch) {
+        onQuerySearch();
+      }
+    }
+  };
+  return (
+    <Box sx={{ pb: 3.75 }} bgcolor="common.white" ref={ref}>
+      <Grid
+        container
+        alignItems="center"
+        justifyContent="space-between"
+        columnGap={1.25}
+        rowGap={1.25}
+      >
+        <Grid item xs={12} lg={2}>
+          <SearchInput
+            value={values.search}
+            onChange={(e) => onChange({ search: e.target.value })}
+            placeholder={searchPlaceholder}
+            onKeyPress={handleKeyPress}
```

Comment:
If not `SearchInput`, then there should be a `ControlledSearchInput` or something that already handle keyPress. Check the code again.